src/main/java/com/vestmark/bitbucket/plugin/MergeConflictDetectorServlet.java

```diff
@@ -82,24 +90,17 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     long pullRequestId = Long.parseLong(urlPathInfoComponents[2]);
     MergeConflictDetector mcd = new MergeConflictDetector(authenticationContext.getCurrentUser(), 
         pullRequestSupplier.getById(repoId, pullRequestId), hostUrl);
-    
-    // If the target is not master, find target branch and upstream releases (if any).
-    if (!mcd.getToBranchId().equals("refs/heads/master")) {
-      String toBranchNamePrefix = mcd.getToBranchName().replaceAll("release\\/.*", "release/");
-      RepositoryBranchesRequest repoBranchesRequest = new RepositoryBranchesRequest
-          .Builder(mcd.getToRepo())
-          .filterText(toBranchNamePrefix)
-          .order(RefOrder.ALPHABETICAL)
-          .build();
-      Page<Branch> branches = refService.getBranches(repoBranchesRequest, 
-          new PageRequestImpl(0, PageRequestImpl.MAX_PAGE_LIMIT));
-      // Iterate over branches and comapre with source branch version to hit upstream releases.
+    BranchClassifier bc = modelService.getModel(mcd.getToRepo()).getClassifier();
+    BranchType toBranchType = bc.getType(mcd.getToBranch());
+    // If the target is not master and is a release branch, find target branch and upstream releases (if any).
+    if (toBranchType.getId().equals("RELEASE") && !mcd.getToBranchId().equals("refs/heads/master")) {
+      Page<Branch> branches = bc.getBranchesByType(toBranchType, new PageRequestImpl(0,PageRequestImpl.MAX_PAGE_LIMIT/2));
       branches.stream()
+              .filter(b -> mcd.isRelated(b))
```

Comment:
minor comment, more for future application, you can use method references here
.filter(mcd::isRelated)